### PR TITLE
Small code fixes accumulated over time

### PR DESCRIPTION
**Pull request by fregante**
Not much, just some changes I excluded from past PRs to keep them focused and easy to review.

Most of these are type-related, mostly fixing `@ts-expect-error` debt

### CODE REVIEW

File: src/background/contentScript.ts [hunk 1]
```diff
@@ -68,36 +69,35 @@ export async function onReadyNotification(signal: AbortSignal): Promise<void> {
   }
 }
 
-// TODO: Use https://github.com/sindresorhus/p-memoize/issues/20 to avoid multiple concurrent calls for every target
 /**
  * Ensures that the contentScript is ready on the specified page, regardless of its status.
  * - If it's not expected to be injected automatically, it also injects it into the page.
  * - If it's been injected, it will resolve once the content script is ready.
  */
-export async function ensureContentScript(
```
Comment (fregante) [inline src/background/contentScript.ts:77]: I just wrapped this in `memoizeUntilSettled`. Tested and it still works

File: src/utils.ts [hunk 2]
```diff
@@ -339,7 +338,7 @@ export function isNullOrBlank(value: unknown): boolean {
 }
 
 export function excludeUndefined(obj: unknown): unknown {
-  if (isPlainObject(obj) && typeof obj === "object") {
+  if (isObject(obj)) {
```
Comment (fregante) [inline src/utils.ts:341]: I wonder if we can replace all or most `_.isPlainObject` calls with our own `isObject` since the former isn't a type guard.

Comment (fregante) [reply src/utils.ts:341]: Ironically this change broke it. Maybe we can just make `isPlainObject` into a type guard so we don't have a redundant `typeof === "object"`

Comment (twschiller) [reply src/utils.ts]: The methods are indeed checking two different things:
- isObject: `typeof === "object"` and not null
- isPlainObject: https://lodash.com/docs/4.17.15#isPlainObject -- has no prototype or the vanilla Object prototype

Agree I think we should be able to override type of isPlainObject to include an assertion

**Review by twschiller — APPROVED** [review]

**Pull request merged**